doc: introduce qml/README.md

jarolrod opened the pull request:
This introduces a `README.md` for the new `qml` directory.  🥳

[qml/README.md render](https://github.com/jarolrod/gui-qml/blob/ef4642c5d51ef6eb3f7de2ec0d246238dc054d8b/src/qml/README.md)

It outlines the goals and limitations of the projects and stages for future development and build notes. 

Looking for concept ACK/NACKS and what's been included and feedback on what else to add. Naturally, this document will evolve as we continue to work on the project.

Closes #1 
Closes #3

hebasto commented on src/qml/README.md:
```diff
@@ -0,0 +1,55 @@
+
+# Development Process
+
+This repo is synced with the [Core repo](https://github.com/bitcoin/bitcoin) on a weekly basis, or as needed to resolve conflicts.
```
Please do not contract "Bitcoin Core" to "Core".

jarolrod replied:
done in https://github.com/bitcoin-core/gui-qml/commit/605842aef74c8c9f79e1fa631b496a991a78a30f

hebasto reviewed (commented):
Concept ACK.

Are we going to define the minimum supported Qt version here on in other pr?

Style nit: not sure if it is worth to use `code formatting` for things other than parts of code or file / directory names.

jarolrod commented:
updated from  ef4642c ->  605842a ([pr7.01](https://github.com/jarolrod/gui-qml/tree/pr7.01) -> [pr7.02](https://github.com/jarolrod/gui-qml/tree/pr7.02))

Changes: Address @hebasto comments

I've removed contractions of `Bitcoin Core` to `Core` and removed the unnecessary occurrences of `Code Formatting`

jarolrod commented:
> Are we going to define the minimum supported Qt version here on in other pr?

Will open a new PR

hebasto reviewed (approved):
ACK 605842aef74c8c9f79e1fa631b496a991a78a30f
